lib/email-validation/blacklisted_email.rb

```diff
@@ -16,19 +12,13 @@ class BlacklistedEmail < ::ActiveRecord::Base
 
     before_save :parse_email
 
-    workflow do
-      state :bounce
-      state :unsubscribe
-      state :admin
-    end
-
     ORIGIN_ADMIN = 'admin'
     ORIGIN_BOUNCE = 'bounce'
     ORIGIN_UNSUBSCRIBE = 'unsubscribe'
 
-    def self.exists? email
+    def self.bounce_or_admin? email
       email = BlacklistedEmail.parse_email_address(email.to_s)
-      BlacklistedEmail.where(email: email).exists?
+      BlacklistedEmail.where(email: email, origin: [ORIGIN_BOUNCE, ORIGIN_ADMIN]).exists?
```

Comment:
Could be but why I'd have to do it differently than the two similar methods that already exists? Besides, there were up to 4 records for the same email in the database and they could have different origin (almost 300 entries with at least duplicated email). Find would find only one, 'where' will get and check the origin for all of them. Completely different story is why our email unique validation doesn't (or didn't) work.